### PR TITLE
#167730733 configure dotenv

### PR DESCRIPTION
**Pull request by OlawaleJoseph**
#### What does this PR do?
  This PR configures the project to set and use environmental variables using dotenv package

#### Description of Task to be completed?
  - Install dotenv package from npm.
  - Create .env, .env_example files
  - Initailize dotenv in the project entry point
  - Document the use of dotenv in the ReadMe file

#### How should this be manually tested?
  - git clone this using the command `git clone git@github.com:andela/firestar-backend.git` in the terminal
  - cd into the project
  - checkout to the branch using the command `git checkout ch-dotenv-setup-167730733`
  - Run the command `npm install` in the terminal
  - Create a `.env` file and input the variable name (uppercase) and value.
       Example PORT=9003
  - In the .env_example file input the name of the variable without a value
      Example PORT= 
  - Open up a terminal and run the command `npm run start:dev`.
  - Check the console and the PORT number specified as the environment variable will be displayed as seen in the image below.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167730733](https://www.pivotaltracker.com/story/show/167730733)

#### Screenshots (if appropriate)
Screenshot (3)

### CODE REVIEW

**Review by marusoft — CHANGES_REQUESTED** [review]
well done @OlawaleJoseph, the following are the changes to be made:
1. provide a link to the branch to be cloned.
2. provide a link project to change directory(cd) to.

**Comment by conquext** [comment]
Clean up the #### in your Readme markup.
image

**Review by marusoft — APPROVED** [review]
well done @OlawaleJoseph

**Comment by conquext** [comment]
Awesome fix. Rebase against changes from merged PRs and resolve conflicts.

**Review by conquext — APPROVED** [review]
Well done!

File: swagger.json [hunk 1]
```diff
@@ -0,0 +1,1035 @@
+{
+  "swagger": "2.0",
+  "info": {
+     "description": "A global Company that make travel and accommodation easy and convenient for the strong workforce of savvy members of staff, by leveraging the modern web.",
+     "version": "1.0.0",
+     "title": "firestar-backend",
```
Comment (rovilay) [inline swagger.json]: @OlawaleJoseph   why is this showing as your changes? Please pull `stage branch` to this branch to keep it up to date.

File: src/index.js [hunk 2]
```diff
@@ -1,93 +1,22 @@
...
-    http = require("http"),
-    path = require("path"),
-    methods = require("methods"),
-    express = require("express"),
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
```
Comment (rovilay) [inline src/index.js]: Please explain why this was removed.

File: src/index.js [hunk 3]
```diff
@@ -1,93 +1,22 @@
...
-
-const isProduction = process.env.NODE_ENV === "production";
+const fs = require('fs'),
+  http = require('http'),
+  path = require('path'),
+  dotenv = require('dotenv')
```
[the remark this answers is not in the document]
Comment (rovilay) [reply src/index.js]: please fix hound issue

File: swagger.json (hunk 1 above)
Comment (OlawaleJoseph) [reply swagger.json]: I have. The branch is up to date

File: src/index.js [hunk 4]
```diff
@@ -1,93 +1,22 @@
...
-    express = require("express"),
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
-
-const isProduction = process.env.NODE_ENV === "production";
+const fs = require('fs'),
```
[the remark this answers is not in the document]
Comment (OlawaleJoseph) [reply src/index.js]: Fixed

File: src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (OlawaleJoseph) [reply src/index.js]: Fixed

File: src/index.js [hunk 5]
```diff
@@ -1,91 +1,14 @@
...
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
+    dotenv =require('dotenv')
+    express = require('express')
+    path = require("path");
 
-const isProduction = process.env.NODE_ENV === "production";
+    // Initiate express App
+    const app = express();
```
[the remark this answers is not in the document]
Comment (OlawaleJoseph) [reply src/index.js]: Fixed

**Pull request merged**